lib/rubocop/cop/rails/root_join_chain.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Use a single `#join` instead of chaining on `Rails.root` or `Rails.public_path`.
+      #
+      # @example
+      #   # bad
+      #   Rails.root.join('db').join('schema.rb')
+      #   Rails.root.join('db').join(migrate).join('migration.rb')
+      #   Rails.public_path.join('path').join('file.pdf')
+      #   Rails.public_path.join('path').join(to).join('file.pdf')
+      #
+      #   # good
+      #   Rails.root.join('db', 'schema.rb')
+      #   Rails.root.join('db', migrate, 'migration.rb')
+      #   Rails.public_path.join('path', 'file.pdf')
+      #   Rails.public_path.join('path', to, 'file.pdf')
+      #
+      class RootJoinChain < Base
+        extend AutoCorrector
+
+        MSG = 'Use `Rails.%<root>s.join(...)` instead of chaining `#join` calls.'
+
+        RAILS_ROOTS = %i[root public_path].to_set.freeze
+
+        RESTRICT_ON_SEND = %i[join].to_set.freeze
+
+        def_node_matcher :rails_root_join_chain, <<~PATTERN
+          (send
+            (send
+              $(send
+                (const {nil? cbase} :Rails) $%RAILS_ROOTS) :join $...) :join $...)
+        PATTERN
+
+        def on_send(node)
+          return unless (rails_root, root, args1, args2 = rails_root_join_chain(node))
```

Comment:
Cosmetic: To my best memory, it's possible to use the block syntax:
```ruby
rails_root_join_chain(node) do |rails_root, root, args1, args2|
  add_offence(...)
```

Reply:
That looks way nicer than the "cloaked" assignment in conditional. Done.